pkg/sqlmigration/049_add_notification_routes.go

```diff
@@ -0,0 +1,290 @@
+			return nil // No rules to migrate
+		}
+		return errors.NewInternalf(errors.CodeInternal, "failed to fetch rules")
+	}
+
+	if len(rules) == 0 {
+		if err := tx.Commit(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	channelsByOrg, err := migration.getAllChannelsInTx(ctx, tx)
+	if err != nil {
+		return errors.NewInternalf(errors.CodeInternal, "fetching channels error: %v", err)
+	}
+
+	var routesToInsert []*expressionRoute
+
+	for _, r := range rules {
+		existingRouteCount, err := tx.NewSelect().
+			Model((*expressionRoute)(nil)).
+			Where("name = ? AND kind = ? AND org_id = ?", r.ID.StringValue(), "rule", r.OrgID).
+			Count(ctx)
+		if err != nil {
+			return errors.NewInternalf(errors.CodeInternal, "failed to check existing route for rule %s", r.ID.StringValue())
+		}
+
+		if existingRouteCount > 0 {
+			continue
+		}
+
+		routeList, err := migration.convertRulesToRoutes([]*rule{r}, channelsByOrg)
+		if err != nil {
+			return errors.NewInternalf(errors.CodeInternal, "converting rules to routes error: %v", err)
+		}
+		if len(routeList) > 0 {
+			routesToInsert = append(routesToInsert, routeList...)
+		}
+	}
+
+	// Insert all routes in a single batch operation
+	if len(routesToInsert) > 0 {
+		_, err = tx.NewInsert().
+			Model(&routesToInsert).
+			Exec(ctx)
+		if err != nil {
+			return errors.NewInternalf(errors.CodeInternal, "failed to insert notification routes")
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (migration *addNotificationRoutes) convertRulesToRoutes(rules []*rule, channelsByOrg map[string][]string) ([]*expressionRoute, error) {
+	var routes []*expressionRoute
+	var errorList []error
+	for _, r := range rules {
+		var gettableRule ruletypes.GettableRule
+		if err := json.Unmarshal([]byte(r.Data), &gettableRule); err != nil {
+			errorList = append(errorList, errors.NewInvalidInputf(errors.CodeInvalidInput, "failed to unmarshal gettableRule for rule %s : error : %v", r.ID.StringValue(), err))
+			continue
+		}
+
+		if len(gettableRule.PreferredChannels) == 0 {
+			channels, exists := channelsByOrg[r.OrgID]
+			if !exists || len(channels) == 0 {
+				continue
+			}
+			gettableRule.PreferredChannels = channels
+		}
+		severity := "critical"
+		if v, ok := gettableRule.Labels["severity"]; ok {
+			severity = v
+		}
+		expression := fmt.Sprintf(`%s == "%s" && %s == "%s"`, "threshold.name", severity, "ruleId", r.ID.String())
+		route := &expressionRoute{
+			identifiable: identifiable{
+				ID: valuer.GenerateUUID(),
+			},
+			timeAuditable: timeAuditable{
+				CreatedAt: r.CreatedAt,
+				UpdatedAt: r.UpdatedAt,
+			},
+			userAuditable: userAuditable{
+				CreatedBy: r.CreatedBy,
+				UpdatedBy: r.UpdatedBy,
+			},
+			Expression:     expression,
+			ExpressionKind: "rule",
+			Channels:       gettableRule.PreferredChannels,
+			Name:           r.ID.StringValue(),
+			Description:    "",
+			Enabled:        !gettableRule.Disabled,
```

Comment:
The Enabled is set based on the rule disabled, however, rule manager doesn't handle the enable/disable to update it. 

Given that enabled is not actually, everything worked without an issue however, if the enabled is used then things break.

And why is that route expression is enabled/disabled based on the rule enabled/disabled?